### PR TITLE
[WIP] Softcut postfilter reset

### PR DESCRIPTION
**Pull request by catfact**
this is a combination of last two PRs (#854, #855) b/c i mixed them together and it didn't seem worth it to untangle them. 

tested on desktop with OSC only. fixed problem with post-filter initialization. sounds nice.

reset seems fine after some more small fixes

### CODE REVIEW

**Comment by catfact** [comment]
still haven't tested any of the lua/c glue, there could be more typos... but pretty striaghtforward

**Comment by tehn** [comment]
awesome. thursday is test day for me, will go through it on hardware.

File: lua/core/softcut.lua [hunk 1]
```diff
@@ -153,39 +172,51 @@ SC.event_phase = function(f) norns.softcut_phase = f end
...
+     state[i].post_filter_hp = 0
+     state[i].post_filter_bp = 0
+     state[i].post_filter_br = 0
+     state[i].level_slew_time = 0.001
+     state[i].rate_slew_time = 0.001
+     state[i].phase_quant = 1
+     state[i].phase_offset = 0
   end
-  SC.buffer_clear()
+  return voice_state
```
Comment (tehn) [inline lua/core/softcut.lua]: typo?

`voice_state` should be `state`?

Comment (catfact) [reply lua/core/softcut.lua]: yaa

**Comment by catfact** [comment]
i would like to make a couple small tweaks/additions that depend on this change but are basically separate. so i'm gonna go ahead and merge it. plz open issues if you find any additional breakage...

**Pull request merged**